cmd/kube-scheduler/app/server.go

```diff
@@ -323,10 +319,14 @@ func Setup(ctx context.Context, opts *options.Options, outOfTreeRegistryOptions
 		scheduler.WithPodInitialBackoffSeconds(cc.ComponentConfig.PodInitialBackoffSeconds),
 		scheduler.WithExtenders(cc.ComponentConfig.Extenders...),
 		scheduler.WithParallelism(cc.ComponentConfig.Parallelism),
+		scheduler.WithComponentConfig(&cc.ComponentConfig),
```

Comment:
All options are about configuring the scheduler, but this one is actually about capturing the output.

So, I would rather call it `WithComponentConfigOutput`. But also we only care about profiles, so could be limited to cc.ComponentConfig.Profiles.

Or you could do `WithCaptureProfiles(func ([]config.KubeSchedulerProfile))` so that the caller is free to do with it whatever they want.

We actually have something like the above that we use for tests https://github.com/kubernetes/kubernetes/blob/master/pkg/scheduler/scheduler.go#L168
It would be cool to unify that. Tests check the expected output, whereas prod scheduler uses it for logs.